v2/pkg/subscraping/types.go

```diff
@@ -20,54 +20,35 @@ type Source interface {
 	// which contains the extractor for subdomains, http client
 	// and other stuff.
 	Run(context.Context, string, *Session) <-chan Result
-	// Name returns the name of the source
+	// Name returns the name of the source. It is preferred to use lower case names.
 	Name() string
+
+	// IsDefault returns true if the current source should be
+	// used as part of the default execution.
+	IsDefault() bool
+
+	// HasRecursiveSupport returns true if the current source
+	// accepts subdomains (e.g. subdomain.domain.tld),
+	// not just root domains.
+	HasRecursiveSupport() bool
+
+	// NeedsKey returns true if the source requires an API key
+	NeedsKey() bool
+
+	AddApiKeys([]string)
 }
 
 // Session is the option passed to the source, an option is created
 // uniquely for each source.
 type Session struct {
 	// Extractor is the regex for subdomains created for each domain
 	Extractor *regexp.Regexp
-	// Keys is the API keys for the application
-	Keys *Keys
```

Comment:
this approach of storing and retreiving keys actually is simpler and easier to maintain compared to the generics one we have implemented now. Any specific reason for the current implementation? This seems to overcomplicate things

Reply:
I strongly disagree. See my comment above :)